platform-sdk/swirlds-config-extensions/src/testFixtures/java/com/swirlds/config/extensions/test/fixtures/ConfigUtils.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.config.extensions.test.fixtures;
+/*
+ * Copyright (C) 2022-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.swirlds.config.api.ConfigData;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.github.classgraph.ClassGraph;
+import io.github.classgraph.ClassInfo;
+import io.github.classgraph.ClassInfoList;
+import io.github.classgraph.ScanResult;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Utility methods for working with configuration.
+ */
+public final class ConfigUtils {
+
+    private ConfigUtils() {}
+
+    /**
+     * Scan all classes on the classpath / modulepath that are under the provided {@code packagePrefixes}.
+     * If the given {@code packagePrefixes} Set is empty all packages will be scanned.
+     *
+     * @param packagePrefixes      the package prefixes to scan
+     * @return the set of all configuration data types found
+     */
+    @SuppressWarnings("unchecked")
+    @NonNull
+    public static Set<Class<? extends Record>> loadAllConfigDataRecords(@NonNull final Set<String> packagePrefixes) {
+        Objects.requireNonNull(packagePrefixes, "packagePrefix must not be null");
+
+        final ClassGraph classGraph = new ClassGraph().enableAnnotationInfo();
+
+        if (!packagePrefixes.isEmpty()) {
+            classGraph.whitelistPackages(packagePrefixes.toArray(new String[0]));
+        }
+
+        try (final ScanResult result = classGraph.scan()) {
```

Comment:
Minor: this may be a heavy operation, especially if certain use cases (e.g. some tests, which is especially important in the context of our goal to speed them up) may not require all the classes to be loaded. Although one could consider it a good "warm up" exercise for the JVM. Also, the classgraph seems to have known issues when running with JDK 16+.

I agree that this scan seems convenient. But perhaps a more optimal solution would be to manually register all the supported/required records somewhere near the application's `main()` method or the tests's `@BeforeAll` method.

Reply:
The only use of this method is to verify that we properly load all platform config data types, which we verify in a single test in `PlatformConfigurationExtensionTests`. This method is not called in production environments or as a regular operation within other unit tests.